lib/compat/wordpress-6.3/get-global-styles-and-settings.php

```diff
@@ -59,16 +45,58 @@ function wp_get_block_css_selector( $block_type, $target = 'root', $fallback = f
 			return $root_selector;
 		}
 
+		$fallback_selector = $fallback ? $root_selector : null;
+
+		// Helper to scope old experimental selectors.
+		$scope_selector = function( $scope, $selector ) {
+			$scopes    = explode( ',', $scope );
+			$selectors = explode( ',', $selector );
+
+			$selectors_scoped = array();
+			foreach ( $scopes as $outer ) {
+				foreach ( $selectors as $inner ) {
+					$outer = trim( $outer );
+					$inner = trim( $inner );
+					if ( ! empty( $outer ) && ! empty( $inner ) ) {
+						$selectors_scoped[] = $outer . ' ' . $inner;
+					} elseif ( empty( $outer ) ) {
+						$selectors_scoped[] = $inner;
+					} elseif ( empty( $inner ) ) {
+						$selectors_scoped[] = $outer;
+					}
+				}
+			}
+
+			return implode( ', ', $selectors_scoped );
+		};
+
+		// Duotone ( may fallback to root selector ).
+		if ( 'filter.duotone' === $target || array( 'filter', 'duotone' ) === $target ) {
+			// If selectors API in use, only use it's value, fallback, or null.
+			if ( $has_selectors ) {
+				return _wp_array_get( $block_type->selectors, array( 'filter', 'duotone' ), $fallback_selector );
+			}
+
+			// Selectors API, not available, check for old experimental selector.
```

Comment:
Is it worth saying that this is left in for backwards compatibility, and should be removed one day?